cf-deploy-teardown/tasks/k8s-ready-state-check.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+#Script to determine is the K8s host is "ready" for cf deployment
+
+set -ex
+
+# cgroup memory & swap accounting in /proc/cmdline
+
+grep -w "cgroup_enable=memory" /proc/cmdline
+echo "Verified: cgroup_enable memory"
+
+grep -w "swapaccount=1" /proc/cmdline
+echo "Verified: swapaccount enabled"
+
+# docker info should show overlay2
+
+docker info | grep -w "Storage Driver: overlay2"
+echo "Verified: 'docker info should show overlay2'"
+
+# kube-dns shows 4/4 ready
+
+kube_dns=$(k get pods : | grep "kube-dns-")
+if [[ $kube_dns == *"4/4 Running"* ]]; then
+  echo "Verified: 'kube-dns shows 4/4 ready'"
+fi
+
+# ntp is installed and running
+
+systemctl status ntpd| grep -w "Active: active (running)"
+echo "Verified: ntp is running"
+
+# "persistent" storage class exists in K8s
+
+k get storageclasses | grep -w "persistent   StorageClass.v1.storage.k8s.io"
+echo "Verified: 'persistent' storage class exists in K8s"
+
+# privileged pods are enabled in K8s
+
+kube_apiserver=$(systemctl status kube-apiserver -l | grep "/usr/bin/hyperkube apiserver" )
+if [[ $kube_apiserver == *"--allow-privileged"* ]]; then
+  echo "Verified: privileged enabled in 'kube-apiserver'"
+fi
+
+kubelet=$(systemctl status kubelet -l | grep "/usr/bin/hyperkube kubelet" )
+if [[ $kubelet == *"--allow-privileged"* ]]; then
+  echo "Verified: privileged enabled in 'kubelet'"
+fi
+
+# dns check for the current hostname resolution
+
+IP=$(nslookup cf-dev.io | grep answer: -A 2 | grep Address: | sed 's/Address: *//g')
```

Comment:
Could this be done now, turning `cf-dev.io` into a variable that's set by the CI system? Ie `${${CF_DEPLOYMENT_HOSTNAME}:-cf-dev.io}`  (I didn't test that, just an idea)